agent/asteroid_agent.py

```diff
@@ -65,15 +66,20 @@ def process(self, message: m.Message) -> None:
                 for target in targets
                 for exploit in self.exploits
             ]
-
             for target_vulnz in futures.as_completed(targets_checks):
-                for vulnerability in target_vulnz.result():
-                    self.report_vulnerability(
-                        entry=vulnerability.entry,
-                        risk_rating=vulnerability.risk_rating,
-                        vulnerability_location=vulnerability.vulnerability_location,
-                        dna=vulnerability.dna,
-                        technical_detail=vulnerability.technical_detail,
+                try:
+                    for vulnerability in target_vulnz.result():
+                        self.report_vulnerability(
+                            entry=vulnerability.entry,
+                            risk_rating=vulnerability.risk_rating,
+                            vulnerability_location=vulnerability.vulnerability_location,
+                            dna=vulnerability.dna,
+                            technical_detail=vulnerability.technical_detail,
+                        )
+                except requests.exceptions.TooManyRedirects as e:
+                    logger.error(
+                        "Too many redirects: %s",
+                        e,
```

Comment:
This should not be here, the redirect is a case of the rule. Rules should check it.

Reply:
In this case, it should be inside the accept method.